src/bda/plone/orders/vocabularies.py

```diff
@@ -96,15 +97,25 @@ def customers_vocab_for(user=None):
             # Development edge case: creator might be None
             continue
         customer = plone.api.user.get(userid=creator)
+
+        email = None
+        name = None
         if customer:
             # soft dep on bda.plone.shop
             first = customer.getProperty('firstname', '')
             last = customer.getProperty('lastname', '')
+            email = customer.getProperty('email', '')
             # fallback
             full = customer.getProperty('fullname', '')
-            name = (first or last) and '{0}, {1}'.format(first, last) or full
+            name = (first or last) and u'{0}, {1}'.format(last, first) or full
```

Comment:
We should also change from the deprecated form of conditional clauses using boolean expression to ternary form.
Instead:
``(first or last) and u'{0}, {1}'.format(last, first) or full``
Better:
``u'{0}, {1}'.format(last, first) if (first or last) else full``

There are some cases where boolean form won't evaluate correctly.